### PR TITLE
LPS-70860 Make all the potential required resources available to the JSP Compiler

### PR DESCRIPTION
**Pull request by migue**

### CODE REVIEW

**Comment by migue** [comment]
The error seems totally unrelated, just running it again to set it to green

ci:retest

**Comment by rotty3000** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by rotty3000** [comment]
hey @migue I had these two PRs in my queue ahead of yours:

https://issues.liferay.com/browse/LPS-69410 | https://github.com/rotty3000/liferay-portal/pull/1284
https://issues.liferay.com/browse/LPS-70245 | https://github.com/rotty3000/liferay-portal/pull/1285

Do you think they were related to this change?

File: modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JspCompiler.java [hunk 1]
```diff
@@ -179,44 +179,50 @@ public void init(
...
+					providedBundleWiring,
+					_collectPackageNames(providedBundleWiring));
+			}
 
-			sb.append("JSP compiler for bundle ");
-			sb.append(bundle.getSymbolicName());
-			sb.append(StringPool.DASH);
-			sb.append(bundle.getVersion());
-			sb.append(" has dependent bundle wirings: ");
+			if (_log.isInfoEnabled()) {
```
Comment (rotty3000) [inline modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JspCompiler.java]: Should this logging block be inside the for loop?

**Comment by rotty3000** [comment]
ci:retest

File: modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JspCompiler.java (hunk 1 above)
Comment (migue) [inline modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JspCompiler.java]: Yes, it should

**Comment by migue** [comment]
Hello @rotty3000 

I am not sure they're related.

This pull solves the scenario where a taglib A uses a class (usually extending it) from a Taglib B (both classes are defined in two separate modules).

Now, a bundle B defines a JSP using the taglib A. Bundle B won't be wired to the module defining the Taglib B, but, however we will need to put this bundle into the set of resources used by the JSP compiler, otherwise, the compiler won't be able to find the taglib B.

I am pretty sure the same scenario will happen if a Taglib A depends on a class C defined in a different module. I haven't solved this one because I wanted to agree with you on this first

File: modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JspCompiler.java (hunk 1 above)
Comment (rotty3000) [inline modules/apps/static/portal-osgi-web/portal-osgi-web-servlet-jsp-compiler/src/main/java/com/liferay/portal/osgi/web/servlet/jsp/compiler/internal/JspCompiler.java]: @migue can I get an updated PR with this change? thx

**Comment by rotty3000** [comment]
Ok, I think I'm ok with this provided we fix the logging block.

**Pull request closed without merging**